File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.py
import sys
input = sys.stdin.readline
from collections import deque

n, k = map(int, input().split())
visited = [0] * (10**5+1)

def find_bro():
    queue = deque([n])
    while queue:
        x = queue.popleft()
        if x == k:
            return
        for dx in [-1, 1, x]:
            nx = x + dx
            if (0<=nx<=10**5) and visited[nx] == 0:
                visited[nx] = visited[x] + 1
                queue.append(nx)
        
find_bro()
print(visited[k])

File: 백준/Silver/1697. 숨바꼭질/숨바꼭질.js
const input = require('fs').readFileSync(process.platform === 'linux' ? '/dev/stdin' : __dirname + '/input.txt').toString().split('\n')

const [n, k] = input[0].split(' ').map(Number)
const MAX_N = 100_000
const visited = Array.from({ length: MAX_N + 1 }, () => -1)

bfs()

console.log(visited[k])

function bfs() {
  let index = 0
  const queue = []

  visited[n] = 0
  queue.push(n)

  while(queue.length) {
    const position = queue[index++]
    if (position === k) return

    for (const d of [- 1, + 1, position]) {
      const next_position = d + position
      if (next_position < 0 || next_position > MAX_N) continue
      if (visited[next_position] !== -1) continue

      queue.push(next_position)
      visited[next_position] = visited[position] + 1
    }
  }
}
